File: snippets/swift/BreezSDKExamples/Sources/LnurlAuth.swift
//
//  LnurlAuth.swift
//
//
//  Created by ruben on 14/11/2023.
//

import BreezSDK
import Foundation

func auth(sdk: BlockingBreezServices) {
    // ANCHOR: lnurl-withdraw
    // Endpoint can also be of the form:
    // keyauth://domain.com/auth?key=val
    let lnurlAuthUrl = "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4excttvdankjm3lw3skw0tvdankjm3xdvcn6vtp8q6n2dfsx5mrjwtrxdjnqvtzv56rzcnyv3jrxv3sxqmkyenrvv6kve3exv6nqdtyv43nqcmzvdsnvdrzx33rsenxx5unqc3cxgeqgntfgu"

    if let inputType = try? parseInput(s: lnurlAuthUrl) {
        if case let .lnUrlAuth(data) = inputType {
            let result = try? sdk.lnurlAuth(reqData: data)
            switch result {
            case .ok:
                print("Successfully authenticated")
            case let .errorStatus(error):
                print("Failed to authenticate: \(error)")
            case .none:
                print("Failed to authenticate")
            }
        }
    }
    // ANCHOR_END: lnurl-withdraw
}

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/LnurlAuth.kt
package com.example.kotlinmpplib

import breez_sdk.*
class LnurlAuth {
    fun lnurlAuth(sdk: BlockingBreezServices) {
        // ANCHOR: lnurl-auth
        // Endpoint can also be of the form:
        // keyauth://domain.com/auth?key=val
        val lnurlAuthUrl = "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7..."
        try {
            val inputType = parseInput(lnurlAuthUrl)
            if (inputType is InputType.LnUrlAuth) {
                when (val result = sdk.lnurlAuth(inputType.data)) {
                    is LnUrlCallbackStatus.Ok -> {} // Log.v("Breez", "Successfully authenticated")
                    is LnUrlCallbackStatus.ErrorStatus -> {} // Log.v("Breez", "Failed to authenticate: ${result.data.reason}")
                }
            }
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: lnurl-auth
    }
}